### DISCUSSION TITLE
Filter active campaigns in report API

### DISCUSSION THREAD

**Discussion by Nilanth** [amzn/ads-advanced-tools-docs#2] [Q&A] [answered]
Hi
I am using `/v2/sp/{recordType}/report` API to get sponsored product report. But I am receiving all campaigns including paused, archived and active. How to filter only active campaigns from this report?

**Comment by elizafarley** [comment, marked as answer]
Try using the `stateFilter` parameter in the request body. `"stateFilter": "enabled"` should return only active campaigns.

**Reply by Nilanth** [reply]
@elizafarley Tried it but seems it's not working...

**Reply by elizafarley** [reply]
Note that the `stateFilter` parameter won't work for an `asins` report, or if you are including the `segment` filter. If you are still seeing paused or archived campaigns despite including `stateFilter`, I'd suggest [opening a support ticket in Jira](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2).
